### ISSUE TITLE
Allow Sending Multi-Line Messages Like  IRC Shift+Enter, in Addition to Riot-Web Shift+Enter, et cetera.

### ISSUE THREAD

**Issue by MilkManzJourDaddy** [matrix-org/matrix-doc#816]
In many IRC clients, one can `<shift>+<enter>` to send a Multi-Line Message.  How this is handled in IRC is that the client sends the messages in order, timed to avoid "flooding".  

Examples of how Matrix Clients currently handle Multi-Line include: 
- On Riot 'droid, one hits `<enter>` to start a new line, twice to start a new paragraph.  The "send" key, obviously, sends the message.  
- In Riot-Web, one does `<shift>+<enter>` to compose a Mulit-Line message. 

This seems related to how messages traverse bridges, i.e. the IRC Bridge, e.g.:  https://github.com/matrix-org/matrix-appservice-irc/issues/366 .

However, a problem in the current state of Chat/IM is ala that old Cingular wireless commercial, where users think "[tisnf](https://searx.itunix.eu?q=%22my%20bff%20jill%22%20AND%20tisnf&categories=general,images)" that other users adhere to traditional language "coding" standards, like the Paragraph, (symbolized by a pilcrow "¶").  Where a concept can't be conveyed by a "wut lol", sentences grouped in paragraphs, separated vertically by a blank line is the convention.  However users complain about that, in general chats.

And, this may not be handled by the IRC Bridge as feeding line-breaks, i.e. html `<br><br>`, with starting a new-line.  

Other Unicode control-characters, or Unicode letter-like characters, depending on the client used, functionally, graphically, work just as one had done `<shift>+<enter>`.  And they may display literally, as possible on the IRC side.  Some Unicode may produce unexpected results.  

What is consistent is that users in large, general rooms, complain.  They sometimes complain about any formatting whatsoever, even quotation marks.

But, composing, and then Cut & Paste, in order to feed each paragraph as a new line allows for messages to send, possibly out of sequence, or out of context where other messages sent in the interim cause confusion.  

### **Any abuses would be handled by Moderation, just as with any other room messages, et cetera.**  

The concept of "Threading" has been mentioned, often.  But the state of Matrix, at present is "Linear", and "Multi-Line" does happen.  So, let's allow them to be handled in a manner which "works".

Either that, or completely remove `<shift>+<enter>` in Riot-Web, along with `<enter>` in Riot 'droid to multi-line, before hitting the `<send-key>`, and then relegate the medium to a series of "wut lol" (see note below) messages, [like the movie within the movie, Idiocracy](https://searx.itunix.eu?q=Idiocracy%20AND%20%22Ass%20the%20Movie%22&categories=general,images).

> Re; "wut lol" above:  The authorship did not anchor fully.  But `<control>+<F>` searching for "The Human Language" within: [10 Things ‘Idiocracy’ Predicted Would Happen, and Sadly Already Have
> "Idiocracy" is coming faster than expected.](http://www.craveonline.com/mandatory/1045965-10-things-idiocracy-predicted-would-happen-and-sadly-already).

**Comment by richvdh** [comment]
What's the TL;DR here? you want to be able to send multiple messages at once instead of a multiple message?

**Comment by MilkManzJourDaddy** [comment]
Well, it's not what I want, but what outspoken users want.  

They don't like what they call, white-space, on their screens.  But to be thorough, inclusive of links, and Copy & Paste, often requires multi-line.  For those who would copy a link, i.e. in Riot-Android, a new line is best.  But, sending the messages line by line, when other traffic intersperses, borks the overall message.  IRC clients feed each `<shift>+<enter> ` line as a new message.  I am fine with the way e.g. Riot-Web `<shift>+<enter> ` works, others are not.  The ones who are fine with that don't speak up.  

Also, related to the above; sometimes hunting down a link takes time, even when not multi-tasking.  Some complain about just dropping a link, where it seems out of context.  And (ro)bots like Ivar may not be online, or respond quickly.  So, you must summarize the context, ala `<shift>+<enter> `.  

Just a note, they do not seem to mind when /var/log content, et cetera, is pasted, white-space and all.

## tl;dr: Give senders a way to break up multi-line`<shift>+<enter> `messages, into multiple messages, posted all together.

**Comment by richvdh** [comment]
If it's just about whitespace, isn't that more about how multi-line messages are formatted?

**Comment by MilkManzJourDaddy** [comment]
Well... I think I have yet to do so, at least on Matrix...  But, other users post bulleted/numbered lists in the rooms.  I see you use them on GitHub.  Those seem to work the same way, or have the same effect, in doing two line-feeds/new-lines, et cetera, to start a new paragraph, and leave a so-called white-space.  At least, it is so, in what I have seen from other users.

You seem quite literate.  I don't always have the perfect grammar, spelling, or syntax, 100% of the time, nor do most of us.  But, I was taught that concepts should be in their own paragraph, and not jumbled together, for clarity, and readability.  

It might very well have been only one user who was rather vocal or outspoken about this, IDK.  But all sorts of things pass in there, including a *problem*, earlier, which was handled. 

People coming over from IRC relate to their experience there.  For example, despite it being part of HomeServer function, they want their own local client logs.  For that, there is an open GitHub issue, plus Matrix Recorder.  But, as Dave wrote, different groups have different expectations.  

For this, maybe it depends on how much of an issue it happens to be.  

Still, as above, simply sending a bulleted/numbered list seems to have the exact same white-space effect, from what I have seen.  

As far as links external to Matrix, at least on mobile, for copying, a separate line is better, white-space or not.  On desktop, it's not so critical.  And, it is moreso a consideration in busy rooms.  But, those have more users, generally, and you can't please everyone.

**Comment by richvdh** [comment]
I don't think there is an issue here

**Comment by gwr** [comment]
Yeah, well, most other chat programs allow a multi-line post, eg.
- MS teams
- Skype
- iMessage
etc.
